realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -396,13 +396,18 @@ private static void initializeSyncedRealm(Realm realm) {
             final RealmProxyMediator mediator = configuration.getSchemaMediator();
             final Set<Class<? extends RealmModel>> modelClasses = mediator.getModelClasses();
 
-            final OsRealmSchema.Creator schemaCreator = new OsRealmSchema.Creator();
-            for (Class<? extends RealmModel> modelClass : modelClasses) {
-                mediator.createRealmObjectSchema(modelClass, schemaCreator);
-            }
+            OsRealmSchema.Creator schemaCreator = new OsRealmSchema.Creator();
+            try {
+                for (Class<? extends RealmModel> modelClass : modelClasses) {
+                    mediator.createRealmObjectSchema(modelClass, schemaCreator);
+                }
 
-            // Assumption: When SyncConfiguration then additive schema update mode.
-            schema = new OsRealmSchema(schemaCreator);
+                // Assumption: When SyncConfiguration then additive schema update mode.
+                schema = new OsRealmSchema(schemaCreator);
+            } finally {
+                schemaCreator.close();
```

Comment:
move this `schemaCreator.close();` to line 443? We have a finally block there already.